codegen/projections/high_score_service/lib/high_score_service/client.rb

```diff
@@ -65,19 +74,22 @@ def initialize(config = HighScoreService::Config.new, options = {})
     #   resp.data.location #=> String
     #
     def create_high_score(params = {}, options = {}, &block)
+      config = options[:plugins] ? @config.dup : @config
+      options[:plugins]&.each { |p| p.call(config) }
```

Comment:
Could we always just dup? Is there a place we can hide this (and other stuff), like maybe a base Client class? Should we type check plugins list here, by forwarding to a plugin list class?

Reply:
I was avoiding the extra cost of copying the object when its not required (ie, no modifications will be made when no operation plugins are used). I have moved it to a private utility method (like apply_middleware) - we can evaluate if all of these should move to a hearth client base in the future.